diff --git a/node_modules/@convex-dev/better-auth/dist/nextjs/index.js b/node_modules/@convex-dev/better-auth/dist/nextjs/index.js
index b2211d7..35845e8 100644
--- a/node_modules/@convex-dev/better-auth/dist/nextjs/index.js
+++ b/node_modules/@convex-dev/better-auth/dist/nextjs/index.js
@@ -9,14 +9,14 @@ const cache = React.cache ||
     });
 const parseConvexSiteUrl = (url) => {
     if (!url) {
-        throw new Error(stripIndent `
+        throw new Error(stripIndent`
       CONVEX_SITE_URL is not set.
       This is automatically set in the Convex backend, but must be set in the Next.js environment.
       For local development, this can be set in the .env.local file.
     `);
     }
     if (url.endsWith(".convex.cloud")) {
-        throw new Error(stripIndent `
+        throw new Error(stripIndent`
       CONVEX_SITE_URL should be set to your Convex Site URL, which ends in .convex.site.
       Currently set to ${url}.
     `);
@@ -42,7 +42,14 @@ export const convexBetterAuthNextJs = (opts) => {
     const siteUrl = parseConvexSiteUrl(opts.convexSiteUrl);
     const cachedGetToken = cache(async ({ forceRefresh } = {}) => {
         const headers = await (await import("next/headers.js")).headers();
-        return getToken(siteUrl, headers, { ...opts, forceRefresh });
+        const mutableHeaders = new Headers();
+        if (headers.has("cookie")) {
+            mutableHeaders.set("cookie", headers.get("cookie"));
+        }
+        if (headers.has("user-agent")) {
+            mutableHeaders.set("user-agent", headers.get("user-agent"));
+        }
+        return getToken(siteUrl, mutableHeaders, { ...opts, forceRefresh });
     });
     const callWithToken = async (fn) => {
         const token = await cachedGetToken();
diff --git a/node_modules/@convex-dev/better-auth/src/nextjs/index.ts b/node_modules/@convex-dev/better-auth/src/nextjs/index.ts
index 0f83886..f03e1bf 100644
--- a/node_modules/@convex-dev/better-auth/src/nextjs/index.ts
+++ b/node_modules/@convex-dev/better-auth/src/nextjs/index.ts
@@ -56,8 +56,8 @@ const nextJsHandler = (siteUrl: string) => ({
 
 type OptionalArgs<FuncRef extends FunctionReference<any, any>> =
   FuncRef["_args"] extends EmptyObject
-    ? [args?: EmptyObject]
-    : [args: FuncRef["_args"]];
+  ? [args?: EmptyObject]
+  : [args: FuncRef["_args"]];
 
 const getArgsAndOptions = <FuncRef extends FunctionReference<any, any>>(
   args: OptionalArgs<FuncRef>,
@@ -74,7 +74,11 @@ export const convexBetterAuthNextJs = (
   const cachedGetToken = cache(
     async ({ forceRefresh }: { forceRefresh?: boolean } = {}) => {
       const headers = await (await import("next/headers.js")).headers();
-      return getToken(siteUrl, headers, { ...opts, forceRefresh });
+      const mutableHeaders = new Headers();
+      headers.forEach((value, key) => {
+        mutableHeaders.append(key, value);
+      });
+      return getToken(siteUrl, mutableHeaders, { ...opts, forceRefresh });
     }
   );
 
